sktime/transformations/series/date.py

```diff
@@ -33,6 +34,15 @@
 ]
 
 
+# TODO: remove in v0.17.0
```

Comment:
the message to the *user* was already ok before this.

You need to write a message to the *developer* that details precisely what needs to be changed when 0.17.0 is released.
Currently, it is wrong, since:
* `deprecated` will raise the message whenever the `DateTimeFeatures` transformer is used, indicating that it will be deprecated as a whole! What you had before, warning in case of parameter set to `True`, was good!
* in the `todo` comment, you should exactly describe what the release manager needs to do - I assume, flip the parameter to `False` and remove the warning? Anything else?